### PR TITLE
Initial commit of heroico recipe.

### PR DESCRIPTION
**Pull request by johnjosephmorgan**
ESPNet recipe for the heroico corpus.

### CODE REVIEW

**Comment by Fhrozen** [comment]
Thank you for your collaboration.

Could you add the information about this corpus on https://github.com/espnet/espnet/blob/master/egs/README.md

File: egs/README.md [hunk 1]
```diff
@@ -18,6 +18,25 @@
 | dirha_wsj               | Distant-speech Interaction for Robust Home Applications      | Multi-Array ASR                            | EN             | https://dirha.fbk.eu/, https://github.com/SHINE-FBK/DIRHA_English_wsj|                               |
 | fisher_callhome_spanish | Fisher and CALLHOME Spanish--English Speech Translation      | ASR/Machine Translation/Speech Translation | ES->EN         | https://catalog.ldc.upenn.edu/LDC2014T23                     |                               |
 | fisher_swbd             | Fisher English Training Speech, Switchboard-1 Release 2      | ASR                                        | EN             | https://catalog.ldc.upenn.edu/LDC2004S13, https://catalog.ldc.upenn.edu/LDC2005S13, https://catalog.ldc.upenn.edu/LDC97S62 |                               |
+
```
Comment (sw005320) [inline egs/README.md]: Some lines are duplicated and the table format is broken.
https://github.com/espnet/espnet/blob/9f3d91ee3996eff1b9fcd6b72510d9b176da9b8f/egs/README.md
Please fix it.

File: egs/heroico/asr1/RESULTS.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+- date: `Mon Dec 30 07:19:17 EST 2019`
+- python version: `3.7.3 (default, Mar 27 2019, 22:11:17)  [GCC 7.3.0]`
+- espnet version: `espnet 0.6.0`
+- chainer version: `chainer 6.0.0`
+- pytorch version: `pytorch 1.0.1.post2`
+- Git hash: `3953bc16899149169a52d55176f0452ee0fd2151`
+  - Commit date: `Fri Dec 27 18:36:25 2019 -0500`
+
+## train_sp_pytorch_train
+### CER
```
Comment (sw005320) [inline egs/heroico/asr1/RESULTS.md]: Can you also add WERs?

File: egs/heroico/asr1/run.sh [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+. ./path.sh || exit 1;
+. ./cmd.sh || exit 1;
+
+# general configuration
+backend=pytorch
+stage=0
+stop_stage=100
+
+ngpu=2 # number of gpus ("0" uses cpu, otherwise use gpu)
```
Comment (sw005320) [inline egs/heroico/asr1/run.sh]: I suggest making `ngpu=1` for a standard recipe.

File: egs/heroico/asr1/run.sh [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+train_dev=devtest
+recog_set="devtest test native nonnative"
+fbankdir=fbank
+lmdatadir=data/local/lm_train
+dict=data/lang_1char/${train_set}_units.txt
+
+
+if [ $stage -le -1 ] && [ $stop_stage -ge -1 ]; then
+    echo "$0 Stage -1: Data download."
+    wget $data_url
```
Comment (sw005320) [inline egs/heroico/asr1/run.sh]: Is it OK to download the LDC data?
Usually, we need to do some process (purchase or register) to obtain the LDC data.

File: egs/heroico/asr1/run.sh [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+    gunzip LDC2006S37.tar.gz
+    tar -xvf LDC2006S37.tar
+fi
+
+if [ $stage -le 0 ] && [ $stop_stage -ge 0 ]; then
+    echo "$0 Stage 0: Data preparation."
+    local/prepare_data.sh $data
+fi
+
+if [ $stage -le 1 ] && [ $stop_stage -ge 1 ]; then
```
Comment (sw005320) [inline egs/heroico/asr1/run.sh]: We prefer to strictly follow the number of stages the same across all recipes for the maintenance purpose.

File: egs/heroico/asr1/run.sh [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+# you can skip this and remove --rnnlm option in the recognition (stage 5)
+if [ -z $lmtag ]; then
+    lmtag=$(basename ${lm_config%.*})
+fi
+lmexpname=train_rnnlm_${backend}_${lmtag}
+lmexpdir=exp/$lmexpname
+mkdir -vp $lmexpdir
+
+if [ $stage -le 8 ] && [ $stop_stage -ge 8 ]; then
```
Comment (sw005320) [inline egs/heroico/asr1/run.sh]: Probably, we can use word-based LM.

File: egs/heroico/asr1/run.sh [hunk 7]
```diff
@@ -52,6 +52,7 @@ set -o pipefail
 
 train_set=train_sp
 train_dev=devtest
+train_test="devtest test native nonnative"
```
Comment (sw005320) [inline egs/heroico/asr1/run.sh]: I think you don't need it.

**Comment by sw005320** [comment]
LGTM!
Once you updated the result, and upload the model, I'll merge it.

**Comment by sw005320** [comment]
Thanks.
WER seems to be broken.
Can you check what happens?

**Comment by johnjosephmorgan** [comment]
Yes, I'll look into it.
Regarding the corpus, we (West Point/ARL) own it. THE LDC told us that
we have the right to post it to openslr.org.

**Comment by sw005320** [comment]
> Yes, I'll look into it. Regarding the corpus, we (West Point/ARL) own it. THE LDC told us that we have the right to post it to openslr.org.

Sounds good.
Thanks for the clarification!

**Comment by johnjosephmorgan** [comment]
No white space between words is being written to the decoder hypotheses.

**Comment by sw005320** [comment]
@johnjosephmorgan, is there any progress?
We'll happy to help your PR.
If you have some issues you could post it or even you can directly email me (shinjiw@ieee.org).

**Comment by johnjosephmorgan** [comment]
Sorry, I've ben away.
I'll get back to this soon.
j

**Pull request closed without merging**